game-core/src/test/java/org/triplea/lobby/common/GameDescriptionTest.java

```diff
@@ -3,46 +3,49 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 
-import java.util.Optional;
+import java.util.Arrays;
+import java.util.Collection;
 
-import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.triplea.game.server.HeadlessGameServer;
 
 import games.strategy.engine.lobby.server.GameDescription;
 
 final class GameDescriptionTest {
-  @Nested
-  final class GetBotSupportEmailTest {
-    @Test
-    void shouldReturnBotSupportEmailWhenBotSupportEmailIsNotEmpty() {
-      final String botSupportEmail = "bot@me.com";
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail(botSupportEmail).build();
-
-      assertThat(gameDescription.getBotSupportEmail(), is(Optional.of(botSupportEmail)));
-    }
-
-    @Test
-    void shouldReturnEmptyWhenBotSupportEmailIsEmpty() {
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail("").build();
-
-      assertThat(gameDescription.getBotSupportEmail(), is(Optional.empty()));
-    }
+  @Test
+  void isBotShouldReturnTrueWhenCommentAndNamePrefixMatch() {
+    final GameDescription gameDescription = GameDescription.builder()
+        .comment(HeadlessGameServer.BOT_GAME_HOST_COMMENT)
+        .hostName(HeadlessGameServer.BOT_GAME_HOST_NAME_PREFIX)
+        .build();
+
+    assertThat(gameDescription.isBot(), is(true));
   }
 
-  @Nested
-  final class IsBotTest {
-    @Test
-    void shouldReturnTrueWhenBotSupportEmailIsNotEmpty() {
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail("bot@me.com").build();
-
-      assertThat(gameDescription.isBot(), is(true));
-    }
-
-    @Test
-    void shouldReturnFalseWhenBotSupportEmailIsEmpty() {
-      final GameDescription gameDescription = GameDescription.builder().botSupportEmail("").build();
+  @ParameterizedTest
+  @MethodSource("notBotDescriptions")
+  void isBotShouldReturnFalseWhen(final GameDescription shouldNotBeBot) {
+    assertThat(shouldNotBeBot.toString(), shouldNotBeBot.isBot(), is(false));
+  }
 
-      assertThat(gameDescription.isBot(), is(false));
-    }
+  private static Collection<GameDescription> notBotDescriptions() {
```

Comment:
Gonna need a `@SuppressWarnings("unused")` here due to the reflective invocation.  (I upgraded to Eclipse 2018-12, and it still generates this warning.)

Reply:
Per: https://github.com/triplea-game/triplea/pull/4545#discussion_r248130230
Rolled back update that added this method: f452e68